roster2/src/java/org/sakaiproject/roster/impl/SakaiProxyImpl.java

```diff
@@ -1160,14 +1160,12 @@ private Cache getCache(String cache) {
             }
 
             Map<String, String> index
-                = (Map<String, String>) cache.get(siteId);
+                = (Map<String, String>) cache.get(siteId+groupId);
 
-            if (index == null) {
-                index = new HashMap<String, String>();
-                for (User user : getSiteUsers(siteId)) {
-                    index.put(user.getDisplayName(), user.getId());
-                }
-                cache.put(siteId, index);
+            if (index == null || index.isEmpty()) {
```

Comment:
You can use `CollectionUtils.isEmpty()' here.

Reply:
Modified!